app/service/sms_sender_rest.py

```diff
@@ -54,7 +56,14 @@ def get_service_sms_senders_for_service(service_id):
 @service_sms_sender_blueprint.route('', methods=['POST'])
 def add_service_sms_sender(service_id):
     form = validate(request.get_json(), add_service_sms_sender_request)
```

Comment:
You haven't made any changes to the schema used to validate the request.  Can the schema not be used to catch invalid data that would result in DataError (below)?

Reply:
Good point! I'll look into that